spec/factories/scheduled_event.rb

```diff
@@ -2,9 +2,50 @@
   factory :scheduled_event do
   	type 'holiday'
   	event_name 'Christmas'
-  	start_time { Date.new(2017, 1, 1) }
-  	one_time true
-  	recurring_rules nil
-  	offset_rule '3'
-  end
+    type 'holiday'
+    event_name 'Christmas'
+    start_time { Date.new(2017, 1, 1) }
+    one_time true
+    offset_rule 3
+
+	  trait :empty_recurring_rules do
+	    recurring_rules nil
+	  end
+
+	  trait :recurring_rules do
+	   recurring_rules { }  
+	  end
+
+	  trait :offset_0 do
+	    offset_rule 0
+	  end
+
+	  trait :offset_1 do
+	    offset_rule 1
+	  end
+
+	  trait :offset_2 do
+	    offset_rule 2
+	  end
+
+	  trait :offset_3 do
+	    offset_rule 3
+	  end
+
+	  trait :offset_4 do
+	    offset_rule 4
+	  end
+
+	  trait :start_on_friday do
```

Comment:
these are better examples of where traits are useful; abstracting away more complex Date logic, etc.